expression/builtin_compare_vec.go

```diff
@@ -278,11 +278,50 @@ func (b *builtinNEIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) e
 }
 
 func (b *builtinLERealSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinLERealSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf0, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err = b.args[0].VecEvalReal(b.ctx, input, buf0); err != nil {
+		return err
+	}
+	buf1, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err = b.args[1].VecEvalReal(b.ctx, input, buf1); err != nil {
+		return err
+	}
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf0, buf1)
+	i64s := result.Int64s()
+	x := buf0.Float64s()
+	y := buf1.Float64s()
+	for i := 0; i < n; i++ {
+		isNull0 := buf0.IsNull(i)
+		isNull1 := buf1.IsNull(i)
+		if isNull0 || isNull1 {
+			if compareNull(isNull0, isNull1) <= 0 {
+				i64s[i] = 1
+			} else {
+				i64s[i] = 0
+			}
+		} else {
+			if types.CompareFloat64(x[i], y[i]) <= 0 {
+				i64s[i] = 1
+			} else {
+				i64s[i] = 0
+			}
+		}
+	}
```

Comment:
We can create a new function about this then all comparison functions like `EQReal`, `GTReal` can use it.